vault/login_mfa.go

```diff
@@ -2013,11 +2015,29 @@ func (c *Core) validateTOTP(ctx context.Context, creds []string, entityMethodSec
 
 	usedName := fmt.Sprintf("%s_%s", configID, creds[0])
 
-	_, ok := c.loginMFABackend.usedCodes.Get(usedName)
+	_, ok := usedCodes.Get(usedName)
 	if ok {
 		return fmt.Errorf("code already used; new code is available in %v seconds", totpSecret.Period)
 	}
 
+	numAttempts, _ := usedCodes.Get(configID)
+	if numAttempts == nil {
+		usedCodes.Set(configID, int64(1), defaultMFAAuthResponseTTL)
```

Comment:
Should we use the `totpSecret.Period` (or `validityPeriod`) here, in the case case an operator sets a shorter or longer TTL?

Reply:
It might also be a good idea to describe this time window to the documentation

Reply:
Ah, yeah, totpSecret.Period is the correct one. Thanks for catching this.
And sure, I will add something to the docs.